scripts/detect_mathlib.sh

```diff
@@ -4,18 +4,11 @@
 echo "::group::Detect Mathlib Output"
 echo "Trying to detect if project is downstream of Mathlib."
 
-# define mathlib dependency patterns for lakefile.lean and lakefile.toml
-dot_lean_pattern="require mathlib"
-dot_toml_pattern="\[\[require\]\]\nname = \"mathlib\"\ngit = \"https://github.com/leanprover-community/mathlib4"
-# Check if lakefile.lean or lakefile.toml contain the mathlib dependency pattern
-if [ -f lakefile.lean ] && grep -q "$dot_lean_pattern" lakefile.lean; then
+# Check if lake-manifest.json contain the mathlib dependency pattern
+if [ -f lake-manifest.json ] && grep -q "mathlib" lake-manifest.json; then
```

Comment:
Is there a misclassification when there is a package Z whose package name contains mathlib, but which does not depend on mathlib, and the repository depends on Z?

Reply:
A command to check dependencies should be implemented in lake, I think.

Reply:
> Is there a misclassification when there is a package Z whose package name contains mathlib, but which does not depend on mathlib, and the repository depends on Z?

Yes, this will misclassify some dependencies. I am going to make this string more specific before merging this. This PR should be in draft.

> A command to check dependencies should be implemented in `lake`, I think.

I agree. I asked about this [here](https://leanprover.zulipchat.com/#narrow/stream/341532-lean4-dev/topic/Lake.20dependency.20graph.20command.20similar.20to.20.60cargo.20tree.60) on Zulip. I think the lake manifest is a better temporary solution that grepping the lake file since in general it will be less brittle and prone to errors like #85.

Reply:
I have now specified the entire mathlib repo URL here

Reply:
Thank you for the review here @Seasawher!